File: VB/MvcDashboardDataSources/Configuration/ExtractDataSourceConfigurator.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWeb
Imports System.Web.Hosting

Namespace MvcDashboardDataSources.Configuration
	Public Class ExtractDataSourceConfigurator
		Public Shared Sub ConfigureDataSource(ByVal configurator As DashboardConfigurator, ByVal storage As DataSourceInMemoryStorage)
			' Registers an Extract data source.
			Dim extractDataSource As New DashboardExtractDataSource("Extract Data Source")
			extractDataSource.Name = "Extract Data Source"
			extractDataSource.ConnectionName = "dataExtract"
			storage.RegisterDataSource("extractDataSource ", extractDataSource.SaveToXml())

			AddHandler configurator.ConfigureDataConnection, AddressOf Configurator_ConfigureDataConnection
		End Sub

		Private Shared Sub Configurator_ConfigureDataConnection(ByVal sender As Object, ByVal e As ConfigureDataConnectionWebEventArgs)
			If e.ConnectionName ="dataExtract" Then
				Dim extractParams As New ExtractDataSourceConnectionParameters()
				extractParams.FileName = HostingEnvironment.MapPath("~/App_Data/SalesPersonExtract.dat")
				e.ConnectionParameters = extractParams
			End If
		End Sub
	End Class
End Namespace

File: CS/MvcDashboardDataSources/Configuration/ExtractDataSourceConfigurator.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWeb;
using System.Web.Hosting;

namespace MvcDashboardDataSources.Configuration {
    public class ExtractDataSourceConfigurator {
        public static void ConfigureDataSource(DashboardConfigurator configurator, DataSourceInMemoryStorage storage) {
            // Registers an Extract data source.
            DashboardExtractDataSource extractDataSource = new DashboardExtractDataSource("Extract Data Source");
            extractDataSource.Name = "Extract Data Source";
            extractDataSource.ConnectionName = "dataExtract";
            storage.RegisterDataSource("extractDataSource ", extractDataSource.SaveToXml());

            configurator.ConfigureDataConnection += Configurator_ConfigureDataConnection;
        }

        private static void Configurator_ConfigureDataConnection(object sender, ConfigureDataConnectionWebEventArgs e) {
            if (e.ConnectionName =="dataExtract") {
                ExtractDataSourceConnectionParameters extractParams = new ExtractDataSourceConnectionParameters();
                extractParams.FileName = HostingEnvironment.MapPath(@"~/App_Data/SalesPersonExtract.dat");
                e.ConnectionParameters = extractParams;
            }
        }
    }
}